Framework/PythonInterface/plugins/algorithms/CompareSampleLogs.py

```diff
@@ -0,0 +1,209 @@
+
+        self.declareProperty(StringArrayProperty(name="SampleLogs", direction=Direction.Input, validator=validator),
+                             doc="Comma separated list of sample logs to compare.")
+        self.declareProperty("Tolerance", 1e-3, validator=FloatBoundedValidator(lower=1e-7, upper=1.0),
+                             doc="Tolerance for comparison of double values.")
+
+        actions = ['warning', 'error']
+        self.declareProperty("DoNotMatchAction", "warning", StringListValidator(actions),
+                             doc="Action to perform if sample logs do not match.")
+        return
+
+    def validateInputs(self):
+        # given workspaces must exist
+        # and must be public of ExperimentInfo
+        issues = dict()
+        workspaces = self.getProperty("InputWorkspaces").value
+        for wsname in workspaces:
+            if not api.AnalysisDataService.doesExist(wsname):
+                issues["InputWorkspaces"] = "Workspace " + wsname + " does not exist."
+            else:
+                wks = api.AnalysisDataService.retrieve(wsname)
+                if isinstance(wks, WorkspaceGroup):
+                    for idx in range(wks.getNumberOfEntries()):
+                        if not isinstance(wks.getItem(idx), MatrixWorkspace):
+                            issues["InputWorkspaces"] = "Group " + wsname + " contains workspaces of unsupported type."
+                elif not isinstance(wks, MatrixWorkspace):
+                    issues["InputWorkspaces"] = "Type of workspace " + wsname + " is not supported by this algorithm."
+
+        return issues
+
+    def _expand_groups(self):
+        workspaces = self.getProperty("InputWorkspaces").value
+        input_workspaces = []
+        for wsname in workspaces:
+            wks = api.AnalysisDataService.retrieve(wsname)
+            if isinstance(wks, WorkspaceGroup):
+                input_workspaces.extend(wks.getNames())
+            else:
+                input_workspaces.append(wsname)
+
+        return input_workspaces
+
+    def compare_properties(self, lhs_run, rhs_run, plist, tolerance=5e-3):
+        """
+        checks whether properties match in the given runs, produces warnings
+            @param lhs_run Left-hand-side run
+            @param rhs_run Right-hand-side run
+            @param plist   List of properties to compare
+        """
+        lhs_title = ""
+        rhs_title = ""
+        if lhs_run.hasProperty('run_title') and rhs_run.hasProperty('run_title'):
+            lhs_title = lhs_run.getProperty('run_title').value
+            rhs_title = rhs_run.getProperty('run_title').value
+
+        # for TOFTOF run_titles can be identical
+        if lhs_title == rhs_title:
+            if lhs_run.hasProperty('run_number') and rhs_run.hasProperty('run_number'):
+                lhs_title = str(lhs_run.getProperty('run_number').value)
+                rhs_title = str(rhs_run.getProperty('run_number').value)
+
```

Comment:
If the workspaces have neither 'run_number' nor 'run_title' logs (is it possible?) lhs_title and/or rhs_title could be empty strings which leads to confusing warning messages.